src/routing_membrane.rs

```diff
@@ -960,26 +960,31 @@ impl<F> RoutingMembrane<F> where F: Interface {
 
         // Temporarily pretend that the sentinel passed, later implement
         // sentinel.
-        let resolved = (SentinelPutRequest::new(message.clone(), data.clone(),
-            our_authority.clone(), source_authority), true);
+        let resolved = Event::PutDataRequest(signed_message.clone(), data.clone(), source_group,
```

Comment:
@mmoadeli, the first argument here should have been the `message.orig_message`. Note this will require change of the types of the first field in each `Event` from `SingedMessage` to `Option<SignedMessage>`.